source/server/options_impl.cc

```diff
@@ -93,24 +93,36 @@ OptionsImpl::OptionsImpl(int argc, char** argv, const HotRestartVersionCb& hot_r
                                              false, ENVOY_DEFAULT_MAX_OBJ_NAME_LENGTH, "uint64_t",
                                              cmd);
 
+  cmd.setExceptionHandling(false);
   try {
     cmd.parse(argc, argv);
   } catch (TCLAP::ArgException& e) {
-    std::cerr << "error: " << e.error() << std::endl;
```

Comment:
Historical fun fact: this was dead code. Since we didn't call `cmd.setExceptionHandling(false)`, TCLAP would always `exit(1)` itself rather than throwing `ArgException` under any circumstances. It's just that our death tests couldn't tell the difference.